consul/consul.go

```diff
@@ -66,7 +67,11 @@ func (r *ConsulAdapter) Register(service *bridge.Service) error {
 func (r *ConsulAdapter) buildCheck(service *bridge.Service) *consulapi.AgentServiceCheck {
 	check := new(consulapi.AgentServiceCheck)
 	if path := service.Attrs["check_http"]; path != "" {
-		check.HTTP = fmt.Sprintf("http://%s:%d%s", service.IP, service.Port, path)
+		formatted_ip := service.IP
+		if net.ParseIP(service.IP).To4() == nil { // verify if its an ipv6 address
+			formatted_ip = fmt.Sprintf("[%s]", service.IP)
```

Comment:
Could you do it without the extra formatted_ip variable, is not needed I think ?
service.IP = fmt.Sprintf("[%s]",service.IP)